algebras/algebra/src/main/scala/endpoints4s/algebra/Urls.scala

```diff
@@ -108,6 +108,24 @@ trait Urls extends PartialInvariantFunctorSyntax {
       value: QueryStringParam[A]
   ): QueryString[A]
 
+  type WithDefault[A]
+
+  /** Builds a `QueryString` with one optional parameter, which has a default value.
+    *
+    * Examples:
+    *
+    * {{{
+    *   optQsWithDefault[Int]("page", 1) // optional `page` parameter, with default value 1
+    * }}}
+    *
+    * @param name Parameter’s name
+    * @tparam A Type of the value carried by the parameter
+    * @group operations
+    */
+  def optQsWithDefault[A](name: String, default: A, docs: Documentation = None)(implicit
```

Comment:
The downside of this approach, compared to #879 is that there is no default implementation for this operator, so binary compatibility of algebra is broken.
(Also there is some repetition in client and server interpreters - basically all client interpreters have the same implementation and all server interpreters have the same implementation, but I'm not sure if that can be improved and code that is repeated is pretty short and simple anyway)

Upside is that it addresses [this comment](https://github.com/endpoints4s/endpoints4s/pull/879#discussion_r735115173).